### ISSUE TITLE
Design a11y and UX issues for clickable teasers

### ISSUE THREAD

**Issue by tonifisler** [vdch/regenair-fabricator#619]
## Description
> Quand je survole les actualités tout au long de la lecture le lecteur d’écran indique que c’est un lien. Est-ce que c’est voulu ? C’est une source de confusion. Dois-je cliquer dessus pour obtenir plus d’information ?

This was reported in the [Accessibility Analysis](https://basecamp.com/1791151/projects/12808497/messages/65330060#comment_497408932)

This is one of the problems for this teaser, there are a number of other UX issues which I will list in the **What we want** section below.

## Problem
News articles (for example) are implemented in a way that make NO sense. You never want to display a whole clickable area when there is **no affordance anywhere that this is a link**. This is simply not accessible in any way. For anybody.

![screen shot 2017-02-07 at 15 59 13](https://cloud.githubusercontent.com/assets/1708450/22696593/dda2d12a-ed4e-11e6-8140-6551aa53adb7.png)

### Screen reader

With a screen reader, you are always reminded that you are in a link, because it is indeed a link. But a link should never wrap so much meaningful content, it's just not semantic. The link should only be on logical bits of information, like the title or image.

### Visually

There is no visual indication that there is a link.

## What we want

We want to be make it accessible for anyone. For that, the design must be thought over and redone. There are a number of things we need in this redesign, and not only regarding the linking issue:

- A proper visually semantic title (news title is more important than the section title "Actualités")
- A logical placement for the category tag (it makes no sense in the image...)
- A Call To Action link, button (something explaining that this is a link to somewhere)
- A proper layout indicating that this is a teaser of another content

## Temporary solution

We could, IMHO, try to improve a11y for people with screen readers as it's at least not a visual change that we can sadly guess will not be improved anytime soon... Let's face it... 😞 

The ideal solution is to have an invisible link overlaying the whole teaser in `position: absolute`

### Currently:

```html
<article class="vd-teaser">
  <a href="..." class="vd-link-invisible">
    <div class="text-center mb-2">
      <h3 class="mb-2">Title</h3>
      <figure class="figure vd-teaser-img">
        <div class="vd-figure-img-container">
          <picture class="figure-img">
            ....
            <img src="..." class="img-fluid">
          </picture>
          <span class="vd-figure-category">Category</span>
        </div>
        <figcaption class="figure-caption">Legend
          <span class="vd-figure-credits">Credits</span>
        </figcaption>
      </figure>
    </div>
    <p>....</p>
    <p class="vd-teaser-meta">2017-02-07</p>
  </a>
</article>
```

### Suggestion

```html
<article class="vd-teaser">
  <div class="text-center mb-2">
    <h3 class="mb-2">Title</h3>
    <figure class="figure vd-teaser-img">
      <div class="vd-figure-img-container">
        <picture class="figure-img">
          ....
          <img src="..." class="img-fluid">
        </picture>
        <span class="vd-figure-category">Category</span>
      </div>
      <figcaption class="figure-caption">Legend
        <span class="vd-figure-credits">Credits</span>
      </figcaption>
    </figure>
  </div>
  <p>....</p>
  <p class="vd-teaser-meta">2017-02-07</p>
  <a href="#" class="vd-teaser-link-invisible"><span class="sr-only">Read more about Title</span></a>
</article>
```

```scss
.vd-teaser {
  position: relative;

  &:hover {
    // whatever hover styles we need
  }
}

.vd-teaser-link-invisible {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}
```

We could even improve the a11y of the whole teaser by wrapping the `<h3>` title in a `<a>` tag.

----

This issue is part of a larger critique I **personally** have with the complete design, which I find to be as anti-web and anti-accessible as can be. This design is not at all what an institution like the Canton de Vaud should be delivering: the main focus should be accessibility, easy access to the content and ease of use. The current design is not providing this. This is a shame, given all the efforts that have been provided in the UX design phase and then in the frontend phase.

**vd.ch should be giving users an easy access to the information they are looking for. Not to a platform with an "original" design where users don't understand anything.**

I have actually heard the word "original" and the expression "sortir des sentiers battus" from the design team. This is not what a website like vd.ch should be focused on.
